### PR TITLE
feat: Support for secret namespace

### PR DESCRIPTION
**Pull request by sabre1041**
### Description
Support for specifying the namespace containing the configuration secret using the format `<namespace>:<name>`

### Checklist
Please make sure that your PR fulfills the following requirements:
- [x] Reviewed the guidelines for contributing to this repository
- [x] The commit message follows the [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [x] Tests for the changes have been updated
- [x] Docs have been added / updated
- [ ] Optional. My organization is added to USERS.md.

### Type of Change

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -17,6 +17,9 @@ metadata:
 type: Opaque
 ```
 You can use it like this: `argocd-vault-plugin generate /some/path -s vault-configuration`.
+
+By default, the secret is assumed to be in the `argocd` namespace. However, the namespace containing the secret can be provided by using the format `<namespaxce>:<name>`
```
Comment (jkayani) [inline docs/config.md:21]:
```suggestion
By default, the secret is assumed to be in the `argocd` namespace. However, the namespace containing the secret can be provided by using the format `<namespace>:<name>` 
```

Comment (sabre1041) [reply docs/config.md]: Fixed

**Review by jkayani — APPROVED** [review]
Thanks a lot - this code looks great, and I appreciate the test added. Also appreciate the attention to backwards compatibility. 

I know #184 aimed to do something similar but doing `$namespace/$name` instead of `$namespace:$name`. It's a small difference, and since this PR is more or less ready now I'll prefer to take this and that one can be updated to change the delimiter when time permits.

**Review by werne2j — APPROVED** [review]
Thanks @sabre1041!

**Comment by sabre1041** [comment]
@jkayani I went back/forth on using a `/` or a `:`. I took the approach that `kubectl create rolebinding` uses when specifying the `--serviceaccount` option

**Comment by redox72** [comment]
Hello, when will this be available in a release? I'm using the argoCD operator and in huge need for this @sabre1041 @werne2j

**Comment by werne2j** [comment]
@redox72 We have a couple more things we want to get into the next release. Will update when we are ready.

**Comment by redox72** [comment]
Thank you for the quick reply. Patiently waiting,

Ido

**Comment by werne2j** [comment]
@redox72 this feature has been released as part of https://github.com/argoproj-labs/argocd-vault-plugin/releases/tag/v1.7.0

**Pull request merged**